### ISSUE TITLE
how to match a transactionID with a transaction from the transaction pool

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain#125]
This was a question coming from the ThreeFold App team. The only answer currently would be to compute it manually based on the transaction received from the transaction pool. This is however not very convenient, given that you now would have to implement the ID logic yourself.

You could argue that this is also desired, given that this means you do not need to trust the explorer, as you would compute it yourself based on the transaction. However, that doesn't get you anywhere given that you anyhow have to trust the explorer that the returned transactions themselves are correct. Therefore there is no extra trust needed to also ensure the transactionID is correct. On top of that we're talking here about unconfirmed transactions, so not even sure it is that important.

A more convenient solution would be that the transactionID is given together with the transaction, when requesting all pending transactions from the transactionpool.

**Comment by LeeSmet** [comment]
imo unless you calculate the transaction ID yourself when submitting the transaction, you can't be sure that the transaction ID will ever be correct if you have no trust relationship with the explorer. We can indeed easily return the ID's as well when checking the transaction pool, but I feel we should at least point out in our docs that anything returned by public explorer can be falsified/faked, and so that they should compute the txn ID themselves. This also allows them to detect malicious (or just broken) explorers by checking if their txn ID is present.

**Comment by GlenDC** [comment]
I agree with you there. But somehow I'm not sure it matters, given the fact that you anyhow have to trust that the transaction data itself is correct, otherwise your computed txnid will also not be correct.

**Comment by LeeSmet** [comment]
> trust that the transaction data itself is correct

Thats my point, the only way to be certain the tx id is valid is to compute it yourself once the transaction is created and *before* you send it to the pool
